### ISSUE TITLE
Output mismatch of torch.max due to extra torch.transpose nodes when running on GPU

### ISSUE THREAD

**Issue by Azyka** [NVIDIA/TensorRT#3451]
## Description

When adding extra nodes of `torch.transpose` as output in this model:
<img width="277" alt="image" src="https://github.com/NVIDIA/TensorRT/assets/74590664/0c4996ad-df39-4c5e-a933-894f5db4cf5a">

New:
<img width="410" alt="image" src="https://github.com/NVIDIA/TensorRT/assets/74590664/886a4022-bf72-4a67-a3eb-5b9aae31c0a2">

The output of `torch.max` is expected to be the same for the same input in this 2 graphs. However, it mismatched bewteew the 2 models.

## Environment

**TensorRT Version**: 8.6.1.post1

**NVIDIA GPU**: RTX 1660

**NVIDIA Driver Version**: 525.147.05

**CUDA Version**: 12.0

**CUDNN Version**: 8.9.4.25

Operating System: Ubuntu 22.04.3 LTS (x86_64)

Python Version (if applicable): 3.10.12

Tensorflow Version (if applicable): 2.13.0

PyTorch Version (if applicable): 2.1.0+cu118

## Relevant Files

**Model link**: 
[models.zip](https://github.com/NVIDIA/TensorRT/files/13343583/models.zip)

Input data file: 
[input_data.zip](https://github.com/NVIDIA/TensorRT/files/13343584/input_data.zip

## Steps To Reproduce

**Script**:
```python
from dataclasses import dataclass
from numpy import testing
import numpy as np
import torch
import tensorrt as trt
import pycuda.driver as cuda
from pycuda.driver import DeviceAllocation
import pickle


@dataclass
class HostDeviceMem:
    host: np.ndarray
    device: DeviceAllocation


class ONNXClassifierWrapper():
    def __init__(self, engine):
        self.engine = engine

    def allocate_memory(self):
        engine = self.engine
        inputs = []
        outputs = []
        bindings = []
        stream = cuda.Stream()
        onames = []
        name2idx = {}
        for idx, binding in enumerate(engine):
            name2idx[binding] = idx
            size = trt.volume(engine.get_binding_shape(binding)) * engine.max_batch_size
            dtype = trt.nptype(engine.get_binding_dtype(binding))
            # Allocate host and device buffers
            host_mem = cuda.pagelocked_empty(size, dtype)
            device_mem = cuda.mem_alloc(host_mem.nbytes)
            # Append the device buffer to device bindings.
            bindings.append(int(device_mem))
            # Append to the appropriate list.
            if engine.binding_is_input(binding):
                inputs.append(HostDeviceMem(host_mem, device_mem))
            else:
                outputs.append(HostDeviceMem(host_mem, device_mem))
                onames.append(binding)
        return inputs, outputs, bindings, stream, onames, name2idx

    def predict(self, inputs):  # result gets copied into output
        (
            trt_inputs,
            trt_outputs,
            trt_bindings,
            stream,
            onames,
            name2idx,
        ) = self.allocate_memory()
        context = self.engine.create_execution_context()
        # print(name2idx)
        # print(inputs.keys())
        for iname in inputs:
            np.copyto(
                trt_inputs[name2idx[iname]].host,
                inputs[iname]
                .astype(trt.nptype(self.engine.get_binding_dtype(iname)))
                .ravel(),
            )

        [cuda.memcpy_htod_async(inp.device, inp.host, stream) for inp in trt_inputs]
        context.execute_async_v2(bindings=trt_bindings, stream_handle=stream.handle)
        [cuda.memcpy_dtoh_async(out.host, out.device, stream) for out in trt_outputs]
        stream.synchronize()
        trt_outputs = [out.host for out in trt_outputs]
        return {
            n: v.reshape(self.engine.get_binding_shape(n))
            for n, v in zip(onames, trt_outputs)
        }


def convert_onnx_to_engine(onnx_filename):
    logger = trt.Logger(trt.Logger.WARNING)
    with trt.Builder(logger) as builder, builder.create_network(
            1 << (int)(trt.NetworkDefinitionCreationFlag.EXPLICIT_BATCH)) as network, \
            trt.OnnxParser(network, logger) as parser:
        # builder.max_workspace_size = max_workspace_size
        # builder.fp16_mode = fp16_mode
        # builder.max_batch_size = max_batch_size

        config = builder.create_builder_config()
        config.set_memory_pool_limit(trt.MemoryPoolType.WORKSPACE, 2 << 30)

        with open(onnx_filename, 'rb') as model:
            if not parser.parse(model.read()):
                for error in range(parser.num_errors):
                    print(parser.get_error(error))

        engine_bytes = builder.build_serialized_network(network, config)

        return trt.Runtime(trt.Logger(trt.Logger.WARNING)).deserialize_cuda_engine(
            engine_bytes
        )

import pycuda.autoinit
class Model0(torch.nn.Module):
    def __init__(self):
        super().__init__()

    def forward(self, *args):
        _args = args
        getitem = _args[0];  _args = None
        clip = torch.clip(getitem, -1, 1)
        cat = torch.cat((clip,), dim = 0)
        max_1 = torch.max(cat, clip)
        return (max_1,)

model_0 = Model0()
output_names_0 = ['v5_0']
input_dict_0 = pickle.load(open('0.pickle', 'rb'))
inputs_0 = tuple(torch.from_numpy(v).to('cuda') for _, v in input_dict_0.items())
torch.onnx.export(model_0, inputs_0, '0.onnx', verbose=False, input_names=['v0_0'], output_names=output_names_0, opset_version=14, do_constant_folding=False)

class Model1(torch.nn.Module):
    def __init__(self):
        super().__init__()

    def forward(self, *args):
        _args = args
        getitem = _args[0]
        transpose = getitem.transpose(1, 0)
        clip = torch.clip(getitem, -1, 1)
        transpose_1 = clip.transpose(0, 1)
        cat = torch.cat((clip,), dim = 0)
        transpose_2 = cat.transpose(0, 1)
        max_1 = torch.max(cat, clip)
        return (transpose, transpose_1, transpose_2, max_1)

model_1 = Model1()
output_names_1 = ['v1_0', 'v2_0', 'v3_0',  'v5_0']
input_dict_1 = pickle.load(open('0.pickle', 'rb'))
inputs_1 = tuple(torch.from_numpy(v).to('cuda') for _, v in input_dict_1.items())
torch.onnx.export(model_1, inputs_1, '1.onnx', verbose=False, input_names=['v0_0'], output_names=output_names_1, opset_version=14, do_constant_folding=False)

engine_0 = convert_onnx_to_engine('0.onnx')
wrapper_0 = ONNXClassifierWrapper(engine_0)
output_0 = wrapper_0.predict(input_dict_0)

engine_1 = convert_onnx_to_engine('1.onnx')
wrapper_1 = ONNXClassifierWrapper(engine_1)
output_1 = wrapper_1.predict(input_dict_1)
output_name_dict = {'v5_0': 'v5_0'}

print('=========================')
try:
    for tensor_name_0, tensor_name_1 in output_name_dict.items():
        testing.assert_allclose(output_0[tensor_name_0], output_1[tensor_name_1], rtol=1, err_msg=f'at {tensor_name_0}, {tensor_name_1}')
    print("tensorRT does not trigger assertion")
except AssertionError as e:
    print("tensorRT triggers assertion")
    print(e)
print('=========================')

```

Steps to repro:
1. Download the input data file and put it at the same dir of the script.
2. Run the script

Output assertion:
```
=========================
tensorRT triggers assertion

Not equal to tolerance rtol=1, atol=0
at v5_0, v5_0
Mismatched elements: 9744 / 9744 (100%)
Max absolute difference: 1
Max relative difference: inf
 x: array([[[[1, 1, 1, ..., 1, 1, 1],
         [1, 1, 1, ..., 1, 1, 1],
         [1, 1, 1, ..., 1, 1, 1],...
 y: array([[[[0, 0, 0, ..., 0, 0, 0],
         [0, 0, 0, ..., 0, 0, 0],
         [0, 0, 0, ..., 0, 0, 0],...
=========================
```

**Have you tried [the latest release](https://developer.nvidia.com/tensorrt)?**: No

**Can this model run on other frameworks?** For example run ONNX model with ONNXRuntime (`polygraphy run <model.onnx> --onnxrt`): Yes

**Comment by Azyka** [comment]
I can't repro in TRT 9.1, seems like fixed in TRT 9.1.

**Comment by zerollzeng** [comment]
Confirm that this has been fix.
```
$ polygraphy run 1.onnx --onnxrt --trt --data-loader-script data_loader.py
[I] RUNNING | Command: /home/scratch.zeroz_sw/miniconda3/bin/polygraphy run 1.onnx --onnxrt --trt --data-loader-script data_loader.py
{'v0_0': array([[[[6, 3, 5, ..., 4, 6, 6],
         [4, 4, 5, ..., 4, 5, 6],
         [7, 4, 6, ..., 6, 6, 5],
         ...,
         [4, 7, 3, ..., 5, 5, 7],
         [6, 7, 4, ..., 7, 5, 4],
         [3, 7, 5, ..., 6, 7, 7]]],


       [[[5, 5, 5, ..., 6, 6, 5],
         [5, 4, 4, ..., 5, 6, 5],
         [5, 6, 6, ..., 4, 4, 7],
         ...,
         [3, 5, 5, ..., 4, 5, 4],
         [7, 5, 6, ..., 6, 6, 7],
         [4, 4, 4, ..., 5, 6, 4]]],


       [[[3, 6, 5, ..., 4, 4, 4],
         [6, 7, 5, ..., 6, 7, 6],
         [7, 3, 4, ..., 6, 4, 6],
         ...,
         [5, 3, 5, ..., 4, 5, 7],
         [3, 6, 4, ..., 4, 4, 4],
         [4, 5, 4, ..., 5, 7, 5]]],


       ...,


       [[[5, 6, 7, ..., 5, 5, 6],
         [4, 4, 6, ..., 6, 5, 6],
         [5, 6, 6, ..., 6, 3, 6],
         ...,
         [5, 5, 7, ..., 4, 3, 3],
         [4, 7, 4, ..., 3, 3, 3],
         [4, 6, 7, ..., 5, 5, 4]]],


       [[[7, 7, 5, ..., 7, 3, 7],
         [4, 5, 4, ..., 7, 4, 4],
         [4, 5, 5, ..., 4, 4, 5],
         ...,
         [6, 6, 5, ..., 5, 7, 6],
         [6, 3, 5, ..., 3, 6, 6],
         [5, 7, 7, ..., 5, 5, 4]]],


       [[[4, 4, 7, ..., 7, 4, 5],
         [5, 5, 7, ..., 5, 4, 7],
         [3, 6, 7, ..., 5, 4, 6],
         ...,
         [5, 5, 6, ..., 6, 4, 6],
         [6, 6, 4, ..., 6, 6, 5],
         [6, 3, 6, ..., 3, 6, 4]]]], dtype=int32)}
[I] onnxrt-runner-N0-11/18/23-10:21:25  | Activating and starting inference
[I] Creating ONNX-Runtime Inference Session with providers: ['CPUExecutionProvider']
[I] onnxrt-runner-N0-11/18/23-10:21:25
    ---- Inference Input(s) ----
    {v0_0 [dtype=int32, shape=(7, 1, 24, 58)]}
[I] onnxrt-runner-N0-11/18/23-10:21:25
    ---- Inference Output(s) ----
    {v1_0 [dtype=int32, shape=(1, 7, 24, 58)],
     v2_0 [dtype=int32, shape=(1, 7, 24, 58)],
     v3_0 [dtype=int32, shape=(1, 7, 24, 58)],
     v5_0 [dtype=int32, shape=(7, 1, 24, 58)]}
[I] onnxrt-runner-N0-11/18/23-10:21:25  | Completed 1 iteration(s) in 0.2291 ms | Average inference time: 0.2291 ms.
[I] trt-runner-N0-11/18/23-10:21:25     | Activating and starting inference
[W] CUDA lazy loading is not enabled. Enabling it can significantly reduce device memory usage and speed up TensorRT initialization. See "Lazy Loading" section of CUDA documentation https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#lazy-loading
[I]     Configuring with profiles: [Profile().add('v0_0', min=[7, 1, 24, 58], opt=[7, 1, 24, 58], max=[7, 1, 24, 58])]
[I] Building engine with configuration:
    Flags                  | []
    Engine Capability      | EngineCapability.DEFAULT
    Memory Pools           | [WORKSPACE: 22551.38 MiB, TACTIC_DRAM: 22551.38 MiB]
    Tactic Sources         | [CUBLAS, CUDNN, EDGE_MASK_CONVOLUTIONS, JIT_CONVOLUTIONS]
    Profiling Verbosity    | ProfilingVerbosity.DETAILED
    Preview Features       | [FASTER_DYNAMIC_SHAPES_0805, DISABLE_EXTERNAL_TACTIC_SOURCES_FOR_CORE_0805]
[I] Finished engine building in 0.169 seconds
[I] trt-runner-N0-11/18/23-10:21:25
    ---- Inference Input(s) ----
    {v0_0 [dtype=int32, shape=(7, 1, 24, 58)]}
[I] trt-runner-N0-11/18/23-10:21:25
    ---- Inference Output(s) ----
    {v1_0 [dtype=int32, shape=(1, 7, 24, 58)],
     v2_0 [dtype=int32, shape=(1, 7, 24, 58)],
     v3_0 [dtype=int32, shape=(1, 7, 24, 58)],
     v5_0 [dtype=int32, shape=(7, 1, 24, 58)]}
[I] trt-runner-N0-11/18/23-10:21:25     | Completed 1 iteration(s) in 0.6652 ms | Average inference time: 0.6652 ms.
[I] Accuracy Comparison | onnxrt-runner-N0-11/18/23-10:21:25 vs. trt-runner-N0-11/18/23-10:21:25
[I]     Comparing Output: 'v1_0' (dtype=int32, shape=(1, 7, 24, 58)) with 'v1_0' (dtype=int32, shape=(1, 7, 24, 58))
[I]         Tolerance: [abs=1e-05, rel=1e-05] | Checking elemwise error
[I]         onnxrt-runner-N0-11/18/23-10:21:25: v1_0 | Stats: mean=5.0135, std-dev=1.226, var=1.5031, median=5, min=3 at (0, 0, 0, 1), max=7 at (0, 0, 0, 14), avg-magnitude=5.0135
[I]         trt-runner-N0-11/18/23-10:21:25: v1_0 | Stats: mean=5.0135, std-dev=1.226, var=1.5031, median=5, min=3 at (0, 0, 0, 1), max=7 at (0, 0, 0, 14), avg-magnitude=5.0135
[I]         Error Metrics: v1_0
[I]             Minimum Required Tolerance: elemwise error | [abs=0] OR [rel=0] (requirements may be lower if both abs/rel tolerances are set)
[I]             Absolute Difference | Stats: mean=0, std-dev=0, var=0, median=0, min=0 at (0, 0, 0, 0), max=0 at (0, 0, 0, 0), avg-magnitude=0
[I]             Relative Difference | Stats: mean=0, std-dev=0, var=0, median=0, min=0 at (0, 0, 0, 0), max=0 at (0, 0, 0, 0), avg-magnitude=0
[I]         PASSED | Output: 'v1_0' | Difference is within tolerance (rel=1e-05, abs=1e-05)
[I]     Comparing Output: 'v2_0' (dtype=int32, shape=(1, 7, 24, 58)) with 'v2_0' (dtype=int32, shape=(1, 7, 24, 58))
[I]         Tolerance: [abs=1e-05, rel=1e-05] | Checking elemwise error
[I]         onnxrt-runner-N0-11/18/23-10:21:25: v2_0 | Stats: mean=1, std-dev=0, var=0, median=1, min=1 at (0, 0, 0, 0), max=1 at (0, 0, 0, 0), avg-magnitude=1
[I]         trt-runner-N0-11/18/23-10:21:25: v2_0 | Stats: mean=1, std-dev=0, var=0, median=1, min=1 at (0, 0, 0, 0), max=1 at (0, 0, 0, 0), avg-magnitude=1
[I]         Error Metrics: v2_0
[I]             Minimum Required Tolerance: elemwise error | [abs=0] OR [rel=0] (requirements may be lower if both abs/rel tolerances are set)
[I]             Absolute Difference | Stats: mean=0, std-dev=0, var=0, median=0, min=0 at (0, 0, 0, 0), max=0 at (0, 0, 0, 0), avg-magnitude=0
[I]             Relative Difference | Stats: mean=0, std-dev=0, var=0, median=0, min=0 at (0, 0, 0, 0), max=0 at (0, 0, 0, 0), avg-magnitude=0
[I]         PASSED | Output: 'v2_0' | Difference is within tolerance (rel=1e-05, abs=1e-05)
[I]     Comparing Output: 'v3_0' (dtype=int32, shape=(1, 7, 24, 58)) with 'v3_0' (dtype=int32, shape=(1, 7, 24, 58))
[I]         Tolerance: [abs=1e-05, rel=1e-05] | Checking elemwise error
[I]         onnxrt-runner-N0-11/18/23-10:21:25: v3_0 | Stats: mean=1, std-dev=0, var=0, median=1, min=1 at (0, 0, 0, 0), max=1 at (0, 0, 0, 0), avg-magnitude=1
[I]         trt-runner-N0-11/18/23-10:21:25: v3_0 | Stats: mean=1, std-dev=0, var=0, median=1, min=1 at (0, 0, 0, 0), max=1 at (0, 0, 0, 0), avg-magnitude=1
[I]         Error Metrics: v3_0
[I]             Minimum Required Tolerance: elemwise error | [abs=0] OR [rel=0] (requirements may be lower if both abs/rel tolerances are set)
[I]             Absolute Difference | Stats: mean=0, std-dev=0, var=0, median=0, min=0 at (0, 0, 0, 0), max=0 at (0, 0, 0, 0), avg-magnitude=0
[I]             Relative Difference | Stats: mean=0, std-dev=0, var=0, median=0, min=0 at (0, 0, 0, 0), max=0 at (0, 0, 0, 0), avg-magnitude=0
[I]         PASSED | Output: 'v3_0' | Difference is within tolerance (rel=1e-05, abs=1e-05)
[I]     Comparing Output: 'v5_0' (dtype=int32, shape=(7, 1, 24, 58)) with 'v5_0' (dtype=int32, shape=(7, 1, 24, 58))
[I]         Tolerance: [abs=1e-05, rel=1e-05] | Checking elemwise error
[I]         onnxrt-runner-N0-11/18/23-10:21:25: v5_0 | Stats: mean=1, std-dev=0, var=0, median=1, min=1 at (0, 0, 0, 0), max=1 at (0, 0, 0, 0), avg-magnitude=1
[I]         trt-runner-N0-11/18/23-10:21:25: v5_0 | Stats: mean=1, std-dev=0, var=0, median=1, min=1 at (0, 0, 0, 0), max=1 at (0, 0, 0, 0), avg-magnitude=1
[I]         Error Metrics: v5_0
[I]             Minimum Required Tolerance: elemwise error | [abs=0] OR [rel=0] (requirements may be lower if both abs/rel tolerances are set)
[I]             Absolute Difference | Stats: mean=0, std-dev=0, var=0, median=0, min=0 at (0, 0, 0, 0), max=0 at (0, 0, 0, 0), avg-magnitude=0
[I]             Relative Difference | Stats: mean=0, std-dev=0, var=0, median=0, min=0 at (0, 0, 0, 0), max=0 at (0, 0, 0, 0), avg-magnitude=0
[I]         PASSED | Output: 'v5_0' | Difference is within tolerance (rel=1e-05, abs=1e-05)
[I]     PASSED | All outputs matched | Outputs: ['v1_0', 'v2_0', 'v3_0', 'v5_0']
[I] Accuracy Summary | onnxrt-runner-N0-11/18/23-10:21:25 vs. trt-runner-N0-11/18/23-10:21:25 | Passed: 1/1 iterations | Pass Rate: 100.0%
[I] PASSED | Runtime: 3.289s | Command: /home/scratch.zeroz_sw/miniconda3/bin/polygraphy run 1.onnx --onnxrt --trt --data-loader-script data_loader.py
```
